dataset/include/scipp/dataset/bins_view.h

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+
+#include "scipp/dataset/dataset.h"
+
+namespace scipp::dataset {
+
+namespace bins_view_detail {
+template <class T, class View> class BinsCommon {
+public:
+  BinsCommon(const View &var) : m_var(var) {}
+  auto indices() const { return std::get<0>(get()); }
+  auto dim() const { return std::get<1>(get()); }
+  auto buffer() const { return std::get<2>(get()); }
+
+protected:
+  auto make(const View &view) const {
+    return make_non_owning_bins(this->indices(), this->dim(), view);
```

Comment:
Defined in `variable/buckets.h` but not explicitly included.

Reply:
Added, though I am not sure what is the right style for template classes.